File: src/immersed.dive.shop.model/Event.cs
using System;
using System.Collections.Generic;

namespace immersed.dive.shop.model;

public class Event : IEntity
{
    public Event()
    {
        Participants = new List<EventParticipant>();
        Dates = new List<EventDate>();
    }

    public virtual List<EventDate> Dates { get; set; }

    public Guid  CourseId { get; set; }
    public virtual Course Course { get; set; }

    public virtual List<EventParticipant> Participants { get; set; }

    public Guid Id { get; set; }
    public DateTime DateCreated { get; set; }
    public DateTime LastUpdated { get; set; }
    public bool Live { get; set; }
}

File: the-client-app/src/app/models/event.ts
import { Course } from "./course";
import { EventDate } from "./eventDate";
import { Participant } from "./participant";

export interface Event {
    id?: string;    
    dates?: EventDate[];
    courseId?: number;
    
    course?: Course;

    participants?: Participant[];

    startDate?: Date;

    dateCreated?: Date;
    lastUpdated?: Date;
    live?:boolean;        
}

export class Event implements Event {
    constructor(event?:Event) {
        if(event) {
            this.id = event.id;
            this.dates = event.dates;
            this.courseId = event.courseId;        
            this.participants = event.participants;
            this.startDate = event.startDate;
            this.dateCreated = event.dateCreated;
            this.lastUpdated = event.lastUpdated;
            this.live = event.live;
        }
    }
}